plaid-stl/src/github/mod.rs

```diff
@@ -385,3 +385,113 @@ pub fn get_repos_for_fpat<T: Display>(
     // to mess with us, this came from a String in the API module.
     Ok(String::from_utf8(return_buffer).unwrap())
 }
+
+pub fn get_branch_protection_rules<T: Display>(
```

Comment:
NIt. We might benefit from you doing a bit of research on what the correct way to do this is. A side effect of the way you've written this is that owner, repo, and branch all must be the same type. I.e (&str, String, String) would not be valid

Reply:
Yeah that's a good point. Will see what I can find here